### PR TITLE
persist date bug

### PR DESCRIPTION
**Pull request by samMcCarthyBJSS**
Story [AB#10110](https://buyingcatalog.visualstudio.com/c5f97979-5b03-4d10-ba8d-871d0526b408/_workitems/edit/10110)
Task [AB#10454](https://buyingcatalog.visualstudio.com/c5f97979-5b03-4d10-ba8d-871d0526b408/_workitems/edit/10454)

### CODE REVIEW

File: app/helpers/routes/getCommencementDate.js [hunk 1]
```diff
@@ -17,5 +17,6 @@ export const getCommencementDate = async ({
   };
 
   const sessionData = { req, key: sessionKeys.plannedDeliveryDate };
-  return getFromSessionOrApi({ sessionData, sessionManager, apiCall });
+  const date = getFromSessionOrApi({ sessionData, sessionManager, apiCall });
+  return date.commencementDate ? date.commencementDate : date;
```
Comment (bjss-chris) [inline app/helpers/routes/getCommencementDate.js:21]: When will `date.commencementDate` be `undefined`?

Comment (samMcCarthyBJSS) [reply app/helpers/routes/getCommencementDate.js:21]: if it gets it from session then it will only have the date as a string

Comment (bjss-chris) [reply app/helpers/routes/getCommencementDate.js:21]: Ah, I see. Might it be worth tweaking `apiCall` to something like this instead? (Caveat: this code may not be 100% correct!)

```javascript
const apiCall = async () => {
    const orgId = req.user.primaryOrganisationId;
    const { commencementDate } = getCommencementDateFromApi({
      orgId,
      accessToken,
      logger,
    });

   return commencementDate;
};
```

What do you think?

Comment (samMcCarthyBJSS) [reply app/helpers/routes/getCommencementDate.js:21]: yeah bit easier than having date.commencementdate to read

**Review by bjss-chris — APPROVED** [review]

**Pull request merged**